src/Cache/Signature.php

```diff
@@ -73,4 +73,19 @@ public function equals(SignatureInterface $signature)
 
         return true;
     }
+
+    private static function utf8Encode(array $data)
+    {
+        if (!function_exists('utf8_encode') || !function_exists('mb_detect_encoding')) {
```

Comment:
the function `utf8_encode` will always exist in this repo, as we polyfill it:
https://github.com/symfony/polyfill-xml/blob/master/bootstrap.php#L14